tutorials/basic-topics/kadane.md

```diff
@@ -0,0 +1,243 @@
+Input: nums = [-2,1,-3,4,-1,2,1,-5,4]
+Output: 6
+Explanation: [4,-1,2,1] has the largest sum = 6.
+```
+
+> A subarray is a contiguous part of an array maintaining the order of elements.            
+>[1,2,3] is a subarray of [1,2,3,4,5] but [1,3,2] & [1,3,5] are not.
+
+Consider an array of positive integers only, the maximum sum subarray will be the entire array itself. If the array contains negative integers only, then the maximum sum subarray will be the maximum element of the array. 
+It gets tricker when we have both positive and negative numbers. 
+
+If the array consists of positive integers(need not be all positive integers). Then there definitely exists a subarray which has positive sum, because I can choose one positive element in worst case and it would be my subarray with positive sum. 
+
+The main idea of Kadane's algorithm is to neglect the negative sum subarrays and take maximum among the positive sum subarrays.
+
+```cpp
+int maxSubArraySum(int arr[], int size)
+{   
+    // maxSum is where the maximum sum of subarray is stored
+    // curSum is where the sum of current subarray is stored
+    int maxSum = INT_MIN, curSum = 0;
+ 
+    for (int i = 0; i < size; i++) {
+        // Add current element to current sum 
+        curSum = curSum + arr[i];
+
+        // If current sum is greater than maxSum, update maxSum
+        if (maxSum < curSum)
+            maxSum = curSum;
+ 
+        // If upon adding ith element current sum is becoming less than 0, it cannot contribute to the maximum sum subarray so we neglect it and reset our current sum to 0 to start another subarray freshly
+        if (curSum < 0)
+            curSum = 0;
+    }
+    return maxSum;
+}
+```
+
+## Steps in Kadane's Algorithm
+
+1. Store the sum of elements of the current subarray found so far. 
+2. Keep updating the maximum sum of subarray found so far by checking if the current sum is greater than the maximum sum found so far or not.
+3. If the current sum is less than 0, then we can neglect the current subarray and start a new subarray from the next element.
+
+export const KadanesuggestedProblems = [
+    {
+        "problemName": "152 - Maximum Product Subarray",
+        "difficulty": "Medium",
+        "leetCodeLink": "https://leetcode.com/problems/maximum-product-subarray/",
+        "solutionLink": ""
+    },
+    {
+        "problemName": "121 - Best Time to Buy and Sell Stock",
+        "difficulty": "Easy",
+        "leetCodeLink": "https://leetcode.com/problems/best-time-to-buy-and-sell-stock/",
+        "solutionLink": ""
+    },
+    {
+        "problemName": "918 - Maximum-sum-circular-subarray",
+        "difficulty": "Medium",
+        "leetCodeLink": "https://leetcode.com/problems/maximum-sum-circular-subarray/",
+        "solutionLink": ""
+    }
+    
+]
+
+<Table title="Suggested Problems" data={KadanesuggestedProblems} />
+
+# Variations of Kadane's Algorithm
+
+## Kadane's 2D Algorithm 
+
+### Overview
```

Comment:
The whole 2D section is rejected.  Already gave you 2 LC problems to walk through. Please redo it.